vocabulary-draft/continuous-deployment-pipeline-events.md

```diff
@@ -0,0 +1,24 @@
+# Continuous Deployment Pipelines Events 
+
+These events are related to continous deployment pipelines and their target environments. 
+These events can be emitted by environments to report where software artifacts such as services, binaries, deamons, jobs or embeded software are running. 
+
+The term Service is used to represent a running Artifact. This service can represent a binary that is running, a deamon, an application, a docker container, etc.
+The term Environment represent any platform which has all the means to run a Service. 
+
+- **Environment Created**: an environment has been created and it can be used to deploy Services
+- **Environment Modified**: an environment has been modified, this event advertise the changes made in the environment
+- **Environment Deleted**: an environment has been deleted and cannot longer be used
+- **Service Deployed**: a new instance of the Service has been deployed
+- **Service Upgraded**: an existing instance of a Service has been upgraded to a new version
+- **Service Undeployed**: an existing instance of a Service has been terminated an it is not longer present in an environment
```

Comment:
This term is a bit tricky. I would agree that "to undeploy" something is to take something that was deployed and make it so that it no longer is deployed.

But "undeployed" would possibly more frequently mean "not deployed yet" (as in "we have 5 undeployed firefighters still available". But probably this is disambiguated by the fact that this is an event, and you wouldn't send an event to announce that something is not deployed yet...

This comment can be ignored :)

Reply:
I do think we should consider this comment because if something can be misunderstood it tends to be misunderstood.

Reply:
@erkist @m-linner-ericsson I am happy to call the event Service Added/Removed for example. Making it explicit that if we are removing something is because it was there. In any case, event systems should deal with those inconsistencies.

Reply:
"undeployed" exists as an adjective in english, and it roughly means "not yet deployed".
"undeploy" as a verb is probably an IT neologism used to express the opposite of "deploy", but I think "removed" or "deleted" might be more common?

Reply:
@afrittoli I've changed it to removed